### PR TITLE
Generate food counts tables

### PR DESCRIPTION
**Pull request by ka-west**

### CODE REVIEW

File: gfop/get_food_counts.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    metadata = pd.read_csv(metadata, sep=delim)
+    metadata = metadata[metadata[filename_col].notnull()]
+    for filename in metadata[filename_col]:
+        file_food_counts = get_file_food_counts(gnps_network, sample_types, all_groups, some_groups, [filename], level)
+        if len(file_food_counts) > 0:
+            food_counts.append(file_food_counts)
+            filenames.append(filename)
+    food_counts = (pd.concat(food_counts, axis=1, sort=True).fillna(0).astype(int).T)
+    food_counts.index = pd.Index(filenames, name='filename')
+    food_counts.to_csv(save_name + '.csv')
```
Comment (bittremieux) [inline gfop/get_food_counts.py]: I'd avoid explicitly saving inside this function. The user can trivially do that themselves afterwards if they want to.

File: gfop/get_food_counts.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                                sample_types = 'simple',
+                                all_groups = ['G1'],
+                                some_groups = ['G4'],
+                                level = 5,
+                                save_name = "food_counts_L5")
+    """
+    food_counts, filenames = [], []
+    gnps_network = pd.read_csv(gnps_network, sep='\t')
+    sample_types = get_sample_types(sample_types)
+    if metadata.endswith('.csv'): delim = ','
```
Comment (bittremieux) [inline gfop/get_food_counts.py]: Weird if-else formatting; normally the statements should be on a new line. A one-liner would be: `delim = ',' if metadata.endswith('.csv') else '\t'

File: gfop/get_food_counts.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    else: delim = '/t'
+    metadata = pd.read_csv(metadata, sep=delim)
+    metadata = metadata[metadata[filename_col].notnull()]
```
Comment (bittremieux) [inline gfop/get_food_counts.py]: Better: `metadata = metadata.dropna(subset=[filename_col])`

**Review by bittremieux — COMMENTED** [review]
Looks pretty good, I've made a few minor comments. There are a few small things that I'd do a bit differently, but no big remarks because in general it's mostly the existing code,

File: gfop/get_food_counts.py (hunk 2 above)
Comment (ka-west) [reply gfop/get_food_counts.py]: Thanks for the tip!

**Comment by ka-west** [comment]
Thanks for the tips and suggestions! Everything has been addressed

**Pull request merged**